docs/en/toolkit/env_config_ref.md

```diff
@@ -0,0 +1,342 @@
+
+Let’s start with a very simple example: an environment in which you have a single engine, with a single app defined within it. The following might be the contents of `project.yml` in this config:
+
+```
+engines:
+  tk-maya:
+    apps:
+      tk-multi-workfiles2:
+        location:
+          type: app_store
+          name: tk-multi-workfiles2
+          version: v0.11.8
+    location:
+        type: app_store
+        name: tk-maya
+        version: v0.9.4
+```
+
+## The engines block
+
+Every environment configuration file starts with an `engines` block. Nested within it are all of the engines defined for that environment.
+
+In our example, only a single engine is defined, `tk-maya`. It has two settings listed, `apps` and `location`. The `apps` setting is a list of all the apps defined for the engine, each with its own settings. In this case, only one app is defined for the engine, `tk-multi-workfiles2`. `location` is a special setting that every bundle requires.
+
+
+## The location descriptor
+
+Every Toolkit bundle has a `location` setting, which we refer to as the bundle’s *descriptor*. The descriptor tells Toolkit where to find the given bundle, and depending on its type, whether to access it directly or cache it locally. Some examples of locations that a Toolkit bundle can come from are the Shotgun App Store, a git repository, a path on disk, or a zip file uploaded to your Shotgun site. Each of these has a corresponding descriptor type, with settings specific to that type. Here again is the descriptor for the `tk-maya` engine from the example above:
+
+```
+    location:
+        type: app_store
+        name: tk-maya
+        version: v0.9.4
+```
+
+This is a descriptor of type `app_store`, which tells Toolkit to get the given bundle from the Shotgun App Store. Descriptors of type `app_store` have the settings `name` and `version`. 
+
+In contrast, if you are actively developing a custom bundle – say you’re working on writing a Toolkit app for a specific workflow in your studio, you may want to get it directly from a path on disk. In this case you’d use a descriptor of type `dev`, which might look like this: 
+
+```
+    location:
+        type: dev
+        path: /path/to/app
+```
+
+The `dev` descriptor has different settings from the `app_store` descriptor. While it can take other settings, it can be set up simply with a `path` setting pointing to where the app lives on disk.
+
+See [the Descriptor section of the Toolkit Core API docs](https://developer.shotgunsoftware.com/tk-core/descriptor.html) for details on all available descriptor types and their settings.
+
+## The apps block
+
+Apps are Toolkit’s user interfaces, and each can run independently of any others. You can choose which apps you want to use based on your pipeline needs, and the `apps` setting within an engine block is where you define which apps are available in a given engine.
```

Comment:
> Apps are Toolkit’s user interfaces, and each can run independently of any others.

I would say this is not completely true. An app is a script/tool that performs an action, they usually have a GUI.
here is an example of an app in our config that doesn't have a GUI and doesn't even register a command.
https://github.com/shotgunsoftware/tk-config-default2/blob/master/env/includes/settings/tk-nuke.yml#L103

Reply:
I didn't say it's graphic, just that it's a user interface – the way users interact with Toolkit. Maybe it'd make more sense to say apps are "user tools"?

Reply:
yeah OK fair enough. As a description it works, though I wonder if there are people that are using the apps for non user interfaces. Probably not, but it's possible. In a way the app I linked isn't ever directly called by a user. And you could automate processes in non user environments that use apps.

Reply:
I changed it to "user tools". I feel like this isn't the place to go into a deep dive on the true definition of an app; the glossary or key concepts/overview would be the spot for that.